libs/partners/cohere/tests/integration_tests/test_chat_models.py

```diff
@@ -61,3 +68,73 @@ def test_invoke() -> None:
 
     result = llm.invoke("I'm Pickle Rick", config=dict(tags=["foo"]))
     assert isinstance(result.content, str)
+
+
+def test_invoke_tool_calls() -> None:
+    llm = ChatCohere(temperature=0)
+
+    class Person(BaseModel):
+        name: str
+        age: int
+
+    tool_llm = llm.bind_tools([Person])
+
+    # where it calls the tool
+    result = tool_llm.invoke("Erick, 27 years old")
+
+    assert isinstance(result, AIMessage)
+    additional_kwargs = result.additional_kwargs
+    assert "tool_calls" in additional_kwargs
+    assert len(additional_kwargs["tool_calls"]) == 1
+    assert additional_kwargs["tool_calls"][0]["function"]["name"] == "Person"
+    assert json.loads(additional_kwargs["tool_calls"][0]["function"]["arguments"]) == {
+        "name": "Erick",
+        "age": 27,
+    }
+
+
+def test_streaming_tool_call() -> None:
+    llm = ChatCohere(temperature=0)
+
+    class Person(BaseModel):
+        name: str
+        age: int
+
+    tool_llm = llm.bind_tools([Person])
+
+    # where it calls the tool
+    strm = tool_llm.stream("Erick, 27 years old")
+
+    additional_kwargs = None
+    for chunk in strm:
+        assert isinstance(chunk, AIMessageChunk)
+        assert chunk.content == ""
+        additional_kwargs = chunk.additional_kwargs
+
+    assert additional_kwargs is not None
+    assert "tool_calls" in additional_kwargs
+    assert len(additional_kwargs["tool_calls"]) == 1
+    assert additional_kwargs["tool_calls"][0]["function"]["name"] == "Person"
+    assert json.loads(additional_kwargs["tool_calls"][0]["function"]["arguments"]) == {
+        "name": "Erick",
+        "age": 27,
+    }
+
+
+def test_streaming_tool_call_no_tool_calls() -> None:
+    llm = ChatCohere(temperature=0)
+
+    class Person(BaseModel):
+        name: str
+        age: int
+
+    tool_llm = llm.bind_tools([Person])
+
+    # where it doesn't call the tool
+    strm = tool_llm.stream("What is 2+2?")
+    acc: Any = None
+    for chunk in strm:
+        assert isinstance(chunk, AIMessageChunk)
+        acc = chunk if acc is None else acc + chunk
+    assert acc.content != ""
+    assert "tool_calls" not in acc.additional_kwargs
```

Comment:
@harry-cohere this is the test that is failing currently.

If it's a model-level problem, I'm happy to skip it.

But currently, this call returns an AIMessage with

- content = ""
- tool calls = []

So basically an empty response :(

Reply:
DMd you! In short, "don't call any tools because they're all irrelevant" is expressed as empty content and an empty tool_calls array. Happy to change to whatever suits the framework! :)

Reply:
Sounds good. I'll mark this test as "skip" for now.

As a heads up, this will make cohere models perform poorly in agents. There's lots of cases, where e.g. you might give an agent a tool like `calculator`, but still want the agent to be able to respond to queries like "what is the capital of france"